lib/redis_memo/memoize_method.rb

```diff
@@ -75,11 +71,20 @@ def self.extract_dependencies(ref, *method_args, &depends_on)
     dependency
   end
 
+  def self.get_or_extract_dependencies(ref, *method_args, &depends_on)
+    if RedisMemo::Cache.local_dependency_cache
+      RedisMemo::Cache.local_dependency_cache[ref] ||= {}
+      RedisMemo::Cache.local_dependency_cache[ref][method_args] ||= extract_dependencies(ref, *method_args, &depends_on)
```

Comment:
We'd need a method idea too! I think we can just use the depends_on object here.

Reply:
Do you mean we can just use the depends_on object as a key? We need to call the depends_on object with the object ref and method args though.

Reply:
Sorry that was not clear at all lol. I think we need to have
```ruby
RedisMemo::Cache.local_dependency_cache[ref][depends_on][method_args]
```
because the same ref might have the same method args for different methods.

Reply:
Ah got it lol, good catch!